src/oci/runc.rs

```diff
@@ -0,0 +1,299 @@
+    /// This can still mean that runc itself failed, which can be verified via the exist status
+    /// of the output.
+    pub async fn resume(&self, args: &[RuncGlobalArgs]) -> Result<Output> {
+        self.exec().run_output(self.binary(), OsStr::new(RUNC_SUB_CMD_RESUME), args.iter().map(ToString::to_string).collect()).await
+    }
+
+    /// Run 'runc run' with the provided args and return the output if the command execution succeeds.
+    /// This can still mean that runc itself failed, which can be verified via the exist status
+    /// of the output.
+    pub async fn run(&self, args: &[RuncGlobalArgs]) -> Result<Output> {
+        self.exec().run_output(self.binary(), OsStr::new(RUNC_SUB_CMD_RUN), args.iter().map(ToString::to_string).collect()).await
+    }
+
+    /// Run 'runc spec' with the provided args and return the output if the command execution succeeds.
+    /// This can still mean that runc itself failed, which can be verified via the exist status
+    /// of the output.
+    pub async fn spec(&self, args: &[RuncGlobalArgs]) -> Result<Output> {
+        self.exec().run_output(self.binary(), OsStr::new(RUNC_SUB_CMD_SPEC), args.iter().map(ToString::to_string).collect()).await
+    }
+
+    /// Run 'runc start' with the provided args and return the output if the command execution succeeds.
+    /// This can still mean that runc itself failed, which can be verified via the exist status
+    /// of the output.
+    pub async fn start(&self, args: &[RuncGlobalArgs]) -> Result<Output> {
+        self.exec().run_output(self.binary(), OsStr::new(RUNC_SUB_CMD_START), args.iter().map(ToString::to_string).collect()).await
+    }
+
+    /// Run 'runc state' with the provided args and return the output if the command execution succeeds.
+    /// This can still mean that runc itself failed, which can be verified via the exist status
+    /// of the output.
+    pub async fn state(&self, args: &[RuncGlobalArgs]) -> Result<Output> {
+        self.exec().run_output(self.binary(), OsStr::new(RUNC_SUB_CMD_STATE), args.iter().map(ToString::to_string).collect()).await
+    }
+    
+    /// Run 'runc update' with the provided args and return the output if the command execution succeeds.
+    /// This can still mean that runc itself failed, which can be verified via the exist status
+    /// of the output.
+    pub async fn update(&self, args: &[RuncGlobalArgs]) -> Result<Output> {
+        self.exec().run_output(self.binary(), OsStr::new(RUNC_SUB_CMD_UPDATE), args.iter().map(ToString::to_string).collect()).await
+    }
+}
+
+#[derive(Clone, Default, Debug)]
+/// DefaultRuncExecCommand is a wrapper which can be used to execute runc in a standard way.
+struct DefaultRuncExecCommand;
+
+impl ExecCommand for DefaultRuncExecCommand {}
+
+#[async_trait]
+trait ExecCommand: Debug + DynClone + Send + Sync {
+    /// Run a command and return its `Output`.
+    async fn run_output(&self, binary: &Path, cmd: &OsStr, args: Vec<String>) -> Result<Output> {
```

Comment:
the sub command can't be prossessed like the args as we don't want the `--` in front of them, but I think it could just be included in the `args` at this point.